Adding PooledByteBufAllocatorMetricSet for Dropwizard metrics

jansyk13 opened the pull request:
JVM memory metric set is not enough for production monitoring(exposes only heap/off heap usage) and there is no way to get more details what Netty does under the hood. This can become handy for example when Ratpack handles large body requests.
Notes:
* adding Dropwizard metric set
* exposing Netty's byte buf allocation metrics
* exposes metrics from PooledByteBufAllocator.DEFAULT because it is used by RatpackDefaultServer
* can expose more detailed metrics(about arenas - chunk of memory allocated using malloc)

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/ratpack/ratpack/1288)
<!-- Reviewable:end -->

johnrengelman commented on ratpack-dropwizard-metrics/src/main/java/ratpack/dropwizard/metrics/DropwizardMetricsModule.java:
```diff
@@ -240,6 +241,18 @@ public void onStart(StartEvent event) throws Exception {
+        // Allocator is resolved in static block in PooledByteBufAllocator based on OS, presence of Unsafe package, etc.
+        metricRegistry.registerAll(
+          new PooledByteBufAllocatorMetricSet(
+            PooledByteBufAllocator.DEFAULT,
```
For some parts of Ratpack, the `ByteBufAllocator` is overridable in the Registry - https://github.com/ratpack/ratpack/blob/master/ratpack-core/src/main/java/ratpack/server/internal/ServerRegistry.java#L98

For example: https://github.com/ratpack/ratpack/blob/master/ratpack-core/src/main/java/ratpack/websocket/WebSockets.java#L77

So, we should probably configure metrics collection for both the `DEFAULT` and the `ByteBufAllocator` from the registry, which you can obtain from the `StartEvent`

johnrengelman commented on ratpack-dropwizard-metrics/src/main/java/ratpack/dropwizard/metrics/DropwizardMetricsConfig.java:
```diff
@@ -62,6 +64,48 @@ public DropwizardMetricsConfig jvmMetrics(boolean jvmMetrics) {
+   * @return True if detailed byte buf allocator metrics collection is enabled. False otherwise
+   * @since 1.6
+   */
+  public boolean isDetailedByteBufAllocatorMetrics() {
```
If there are multiple configuration options for this feature, then we should create a `ByteBufAllocatorConfig` object and nest the settings together rather than exposing both at the top level of `DropwizardMetricsConfig`.

See how `JmxConfig` is done in this class.

jansyk13 commented:
@johnrengelman Hey I fixed your comments. The bb allocator is now injected from registry, added metric set for unpooled allocator and added config class.
